OpenRA.Game/Game.cs

```diff
@@ -776,9 +784,17 @@ static void Loop()
 
 			while (state == RunStatus.Running)
 			{
-				// Ideal time between logic updates. Timestep = 0 means the game is paused
+				// Ideal time between logic updates. ReplayTimestep = 0 means the replay is paused
 				// but we still call LogicTick() because it handles pausing internally.
-				var logicInterval = worldRenderer != null && worldRenderer.World.Timestep != 0 ? worldRenderer.World.Timestep : Ui.Timestep;
+				var logicInterval = Ui.Timestep;
+				if (worldRenderer?.World != null)
+				{
+					var world = worldRenderer.World;
+					logicInterval = world.IsLoadingGameSave ? 1 :
+						world.IsReplay && world.ReplayTimestep != 0 ? world.ReplayTimestep :
+						world.IsReplay && world.ReplayTimestep == 0 ? Ui.Timestep :
```

Comment:
Style nit: my previous suggestion here was explicitly aiming to avoid reassigning `UI.Timestep` to the variable again. I agree that this case does seem more correct to use `Ui.Timestep`, but it does mess up my original idea.

How about changing this to
```csharp
var logicInterval = Ui.Timestep;
var logicWorld = worldRenderer?.World;

// ReplayTimestep = 0 means the replay is paused: we need to keep logicInterval as UI.Timestep to avoid breakage
if (logicWorld != null && !(logicWorld.IsReplay && logicWorld.ReplayTimestep == 0))
	logicInterval = logicWorld.IsLoadingGameSave ? 1 :
		logicWorld.IsReplay ? logicWorld.ReplayTimestep :
		logicWorld.Timestep;
```
instead?